src/Microsoft.Extensions.Http/HttpMessageHandlerPolicyCollection.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Microsoft.Extensions.Http
+{
+    public class HttpMessageHandlerPolicyCollection : Collection<IHttpMessageHandlerPolicy>
+    {
+        public IList<string> Stages { get; } = new List<string>();
```

Comment:
This could probably be it's own list on options. I wanted to see what would happen if I did this, now I realise that it's **super** wacky #YOLO